Common/SymbolCache.cs

```diff
@@ -115,6 +130,41 @@ public static void Clear()
             _cache = new Cache();
         }
 
+        private static Tuple<bool, Symbol, InvalidOperationException> TryGetSymbol(string ticker)
+        {
+            Symbol symbol;
+            InvalidOperationException error = null;
+            if (!_cache.TryGetSymbol(ticker, out symbol))
+            {
+                // fall-back full-text search as a back-shim for custom data symbols.
+                // permitting a user to use BTC to resolve to BTC.Bitcoin
+                var search = $"{ticker.ToUpperInvariant()}.";
+                var match = _cache.Symbols.Where(kvp => kvp.Key.StartsWith(search)).ToList();
+
+                if (match.Count == 0)
+                {
+                    // no matches
+                    error = new InvalidOperationException($"We were unable to locate the ticker '{ticker}'.");
+                }
+                else if (match.Count == 1)
+                {
+                    // exactly one match
+                    symbol = match.Single().Value;
+                }
+                else if (match.Count > 1)
+                {
+                    // too many matches
+                    error = new InvalidOperationException(
```

Comment:
What would happen if we have `"SPY"` and `"SPY.MyCustomData"`, will the user be able to use `"SPY"` from the cache? Specifically thinking if it will cause this exception to be thrown
I think it would be a good idea to add an algorithm which showcases this issue, has a collision, and what is the expected behavior

Reply:
Since `"SPY"` is a direct match, it will resolve properly. If there isn't already a unit test for this exact case, I'll be sure to add one.

Reply:
Added the following unit test:
``` csharp
[Test]
public void TryGetSymbol_FromTicker_WithConflictingSymbolWithCustomDataSuffix()
{
    var ticker = "My-Ticker";
    var symbol = Symbol.Create(ticker, SecurityType.Equity, Market.USA);
    SymbolCache.Set(ticker, symbol);

    var customTicker = $"{ticker}.CustomDataTypeName";
    var customSymbol = Symbol.Create(customTicker, SecurityType.Base, Market.USA, baseDataType: typeof(Bitcoin));
    SymbolCache.Set(customTicker, customSymbol);

    Symbol fetchedSymbol;
    Assert.IsTrue(SymbolCache.TryGetSymbol(ticker, out fetchedSymbol));
    Assert.AreEqual(symbol, fetchedSymbol);

    Symbol fetchedCustomSymbol;
    Assert.IsTrue(SymbolCache.TryGetSymbol(customTicker, out fetchedCustomSymbol));
    Assert.AreEqual(customSymbol, fetchedCustomSymbol);
}
```

Now... you bring up a _really_ great point. Consider `AddData<T>("SPY")` and `AddEquity("SPY")` -- using the `string` `"SPY"` will _always_ resolve to the equity and _never_ to the custom data type, which would require you to specify `"SPY.TypeName"`. This is another potentially breaking change, although I'm not sure how common this collision is in practice.

In the case where there is no potential for collision, say `AddData<T>("SPY")` and `AddEquity("Not-SPY")`, using `string` `"SPY"` will resolve to the only potential match, which would be `"SPY.TypeName"` since there is not entry that exactly matches `"SPY"`.